src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPanelMapping.java

```diff
@@ -57,12 +58,39 @@ public static List<String> getAuthorisationSubTypes() {
         return Collections.emptyList();
     }
 
-    public static List<PanelPreference> getPanelPreferences() {
-        // TODO Adjournments - loop to go through Judicial members that are need to be included or excluded
-        // TODO Potentially used with Manual overrides
-        //      Will need Judicial Staff Reference Data
+    public static List<PanelPreference> getPanelPreferences(SscsCaseData caseData,
+                                                            ReferenceDataServiceHolder referenceDataServiceHolder) {
+        Adjournment adjournment = caseData.getAdjournment();
+        ArrayList<PanelPreference> panelPreferences = new ArrayList<>();
+
+        if (referenceDataServiceHolder.isAdjournmentFlagEnabled()
+            && isYes(adjournment.getAdjournmentInProgress())) {
+            panelPreferences.addAll(getAdjournmentPanelPreferences(adjournment.getPanelMembers()));
+
+            AdjournCasePanelMembersExcluded panelMembersExcluded = adjournment.getPanelMembersExcluded();
+            if (panelMembersExcluded == AdjournCasePanelMembersExcluded.YES) {
+                panelPreferences.forEach(panelPreference -> panelPreference.setRequirementType(RequirementType.EXCLUDE));
+            } else if (panelMembersExcluded == AdjournCasePanelMembersExcluded.RESERVED) {
+                panelPreferences.forEach(panelPreference -> panelPreference.setRequirementType(RequirementType.MUST_INCLUDE));
+            }
+        }
+
+        return  panelPreferences;
+    }
```

Comment:
Creation of the initial ArrayList can be avoided like this. The peek method in stream api is an intermediate operation that allows you to perform an action on each element of the stream as they are consumed from the resulting stream. It doesn't modify the original elements, but you can use it to perform some side-effect actions, such as changing the state of the elements. As it is an intermediate operation, it will not be executed until a terminal operation is called on the stream, ie collect.

```suggestion
public static List<PanelPreference> getPanelPreferences(SscsCaseData caseData, ReferenceDataServiceHolder referenceDataServiceHolder) {
    Adjournment adjournment = caseData.getAdjournment();

    if (referenceDataServiceHolder.isAdjournmentFlagEnabled() && isYes(adjournment.getAdjournmentInProgress())) {
        List<PanelPreference> panelPreferences = getAdjournmentPanelPreferences(adjournment.getPanelMembers());
        AdjournCasePanelMembersExcluded panelMembersExcluded = adjournment.getPanelMembersExcluded();

        if (panelMembersExcluded == AdjournCasePanelMembersExcluded.YES) {
            return panelPreferences.stream()
                .peek(panelPreference -> panelPreference.setRequirementType(RequirementType.EXCLUDE))
                .collect(Collectors.toList());
        } else if (panelMembersExcluded == AdjournCasePanelMembersExcluded.RESERVED) {
            return panelPreferences.stream()
                .peek(panelPreference -> panelPreference.setRequirementType(RequirementType.MUST_INCLUDE))
                .collect(Collectors.toList());
        }

        return panelPreferences;
    }

    return new ArrayList<>();
}

```